backend/migrations/dataall_migrations/base_migration.py

```diff
@@ -0,0 +1,58 @@
+def protect(*protected):
+    """Returns a metaclass that protects all attributes given as strings"""
+
+    class Protect(type):
+        has_base = False
+
+        def __new__(meta, name, bases, attrs):
+            if meta.has_base:
+                for attribute in attrs:
+                    if attribute in protected:
+                        raise AttributeError('Overriding of attribute "%s" not allowed.' % attribute)
+            meta.has_base = True
+            klass = super().__new__(meta, name, bases, attrs)
+            return klass
+
+    return Protect
+
+
+class BaseDataAllMigration(metaclass=protect('set_next', 'set_previous', 'is_initial', 'is_last')):
+    key = '-1'
+    name = 'Base Migration'
+    description = 'Base Migration'
+
+    @classmethod
+    def up(cls):
+        print(f'Upgrade is not defined for migration {cls.name}')
+
+    @classmethod
+    def down(cls):
+        print(f'Downgrade is not defined for migration {cls.name}')
```

Comment:
I think this class is mixing management and execution/business logic. 

I think this should be an `ABC` with abstact props/methods `name` `description` `up` `down`  which will force all derived classes to implement them. Similarly to alembic instead of key I'd have a `previous_revision` and `next_revision`.

Then the herder/management should deal with the execution order much easier than it currently does.


Wdyt?

Reply:
I thought about bi-directional links to previous and next revisions.
I postponed it, because now the keys are appointed manually, so I don't want developer to do more actions, than necessary. I'm not sure about our timeline, so I don't know, if I can come back to this any time soon. 
Since, there are only one migration now and I don't think we will have many more, I would leave it for further enhancements. 
Of course, if you think, that the current solution is not good enough for main, I can fix it)) But I don't want to postpone this PR, because we are close to 2.6.

Reply:
Thinking about it we don't even need bidirectional since we probably don't have a scenario that we will want to go to the previous node. 
I think such an implementation (simple single linked list traversal) is quite easy to implement, do you think you can do it in the current timeframe?

Reply:
I made the BaseMigration ABC